scripts/src/java/org/oppia/android/scripts/common/BazelClient.kt

```diff
@@ -57,16 +54,19 @@ class BazelClient(
     // Note that this check is needed since rbuildfiles() doesn't like taking an empty list.
     return if (buildFileList.isNotEmpty()) {
       val referencingBuildFiles =
-        executeBazelCommand(
-          "query",
+        runPotentiallyShardedQueryCommand(
+          "filter('^[^@]', rbuildfiles(%s))", // Use a filter to limit the search space.
```

Comment:
TODO: Mention this in PR description. It's a significant reason why ComputeAffectedTests now works correctly for very large PR results since this was found to be the primary bottleneck in performance. Sharding should be more or less the same in behavior (see comment below).

Reply:
Done.